### PR TITLE
Copy Target's Mount

### PR DESCRIPTION
**Pull request by TylerR909**
This was a feature I enjoyed from the now-defunct Pets & Mounts addon. I'm thoroughly enjoying BeStride as a replacement for PAM and would love to contribute this feature to the addon.

I did find an old issue requesting it: https://github.com/Bestride/BeStride/issues/128

~~From my testing this should be fine as-is for Retail, but after I pull myself out of the rabbit hole that is DF Launch Week I'd be happy to take a crack at WotLK and isolating certain helpers behind Versions/Common or Versions/Mainline~~ edit: Updated to support WotLK

### CODE REVIEW

File: Versions/Common/mount.lua [hunk 1]
```diff
@@ -264,3 +267,17 @@ function BeStride:DismountAndExit()
 		Dismount()
 	end
 end
+
+function BeStride:UseTargetsMount()
+	local mountSpellId = nil
+	AuraUtil.ForEachAura("target", "HELPFUL", 40, function(...)
+		local spellId = select(10, ...)
+		local spellIsMount = C_MountJournal.GetMountFromSpell(spellId)
```
Comment (DanSheps) [inline Versions/Common/mount.lua:275]: Same as above, I will develop a helper for this and let you know

File: Versions/Common/logic.mounts.lua [hunk 2]
```diff
@@ -52,4 +52,24 @@ function BeStride:CheckLoanedMount()
...
 	
 	return nil
+end
+
+function BeStride:CanUseTargetsMount()
+	if (not self:DBGet("settings.mount.copytargetmount") or not BeStride:IsMainline()) then return false end
+	if (self:MovementCheck() or not UnitExists("target")) then return false end
+
+	local mountId = nil
+	AuraUtil.ForEachAura("target", "HELPFUL", 40, function(...)
```
Comment (TylerR909) [inline Versions/Common/logic.mounts.lua:62]: It doesn't look like `AuraUtil.ForEachAura` is defined in WotLK so I may have to switch to manually looping through `UnitBuff("target",index)` for that (or both).

File: Versions/Common/logic.mounts.lua [hunk 3]
```diff
@@ -52,4 +52,24 @@ function BeStride:CheckLoanedMount()
...
+		local spellId = select(10, ...)
+		local spellIsMount = C_MountJournal.GetMountFromSpell(spellId)
```
[the remark this answers is not in the document]
Comment (TylerR909) [reply Versions/Common/logic.mounts.lua:64]: Looks like you may already have everything we need for it.

```lua
function BeStride:IsSpellMount(spellId)
  return BeStride_MountDB[spellId] ~= nil
end
```

I'd be happy to contribute these but I'll be wrapped up in DF launch for the next 2-3 days minimum. I am active in WotLK so I can help test over there too.

[the remark this answers is not in the document]
Comment (DanSheps) [reply Versions/Common/logic.mounts.lua:64]: I think that only works for wrath.

**Comment by TylerR909** [comment]
@DanSheps I did a pretty decent refactor which isolated "Get Known Mount From Target" into client-specific helpers that both return `spellId, mountId, isUsable`. Now `CanUseTargetsMount` and `UseTargetsMount` rely on a shared helper and are basically 2-liners each.

I saw some of the new functions come in but I don't think we need those anymore. Unless you have a reason to keep them around, I think we can go ahead and nix them.

**Comment by TylerR909** [comment]
Also worth noting I tested this in Retail and Wrath.

Let me just say, _finding_ someone at 3am on a weeknight in Dalaran who has a mount you have is a huge pain. But, I only favorited a 60% and he was walking around on a 100% I knew and Regular Mount selected the 100% while I had him selected, and my favorited 60% any time I clicked off him.

File: Versions/Common/mount.lua [hunk 4]
```diff
@@ -1,6 +1,9 @@
 function BeStride:Regular()
 	-- Aspect of the Cheetah is available from level 5
```
Comment (TylerR909) [inline Versions/Common/mount.lua:2]: I should move this comment down

File: Versions/Wrath/bestride.lua [hunk 5]
```diff
@@ -50,4 +50,20 @@ function BeStride:GetMountInfoByIndex(index)
...
+end
+
+function BeStride:GetKnownMountFromTarget()
+	local mountIdBySpellId = {}
+	for i=1,GetNumCompanions("MOUNT"),1 do
+		local mountID,name,spellID,icon,isSummoned = GetCompanionInfo("MOUNT", i)
+		mountIdBySpellId[spellID] = mountID
+	end
+	-- look for unit aura that matches known AND usable mount ID
+	local mountId = nil
```
Comment (TylerR909) [inline Versions/Wrath/bestride.lua:62]: unused

**Comment by DanSheps** [comment]
FYI, I was away all of December and into early Jan.  I will look at this shortly.

**Comment by TylerR909** [comment]
All of these open PRs have been merged on my local fork since opening them here. All working great for the last 1-2 months.

**Review by DanSheps — APPROVED** [review]

**Review by DanSheps — APPROVED** [review]

**Comment by DanSheps** [comment]
Merged

**Pull request closed without merging**